test/SdkCoinTest.t.sol

```diff
@@ -4,385 +4,602 @@ pragma solidity >=0.8.25 <0.9.0;
+        } else if (convertedAmount < SafeCast.toUint64(amount)) {
+            invariant_ERC20toSdkCoin_RemainderByFactor(amount, remainder, factor);
+        }
     }
 
-    function testConvertSdkCoinAmountToERC20_LessThanSixDecimals() public {
-        MockERC20Metadata customMockERC20Metadata = new MockERC20Metadata(5);
-        uint64 cosmosAmount = 1_000_000; // 1 Cosmos coin
-
-        vm.expectRevert(abi.encodeWithSelector(IISdkCoinErrors.UnsupportedTokenDecimals.selector, uint8(5)));
-        SdkCoin._convertSdkCoinAmountToERC20(address(customMockERC20Metadata), cosmosAmount);
+    //////////////////////////////////////
+    // Invariants Tests
+    // Invariant test: ERC20 -> SdkCoin with tokenDecimals == DEFAULT_COSMOS_DECIMALS,
+    /**
+     * @notice Invariant test for ERC20 to Cosmos coin conversion when token decimals are equal to Cosmos decimals
+     * @notice Given random inputs, this test applies invariant checking to ensure the converted amount equals the input
+     * amount and the remainder is zero
+     * @param amount The amount of ERC20 tokens to be converted
+     * Requirements:
+     * - `amount` must be greater than 0 and less than or equal to the maximum uint64 value.
+     */
+    function testInvariant_ERC20toSdkCoin_EqualDecimals(uint256 amount) public {
+        // Skip test for zero amount or overflow conditions
+        if (amount == 0 || amount > ~uint64(0)) {
+            // These conditions will revert and are expected behaviours that have already been covered in table testing
+            return;
+        }
+        address tokenAddress = address(new MockERC20Metadata(6));
+        (uint64 convertedAmount, uint256 remainder) =
+            SdkCoin._ERC20ToSdkCoin_ConvertAmount(address(tokenAddress), amount);
+        assertInvariants_ERC20toSdkCoin_EqualDecimals(convertedAmount, amount, remainder);
     }
 
-    function testFailConvertSdkCoinAmountToERC20_ZeroAmount() public {
-        MockERC20Metadata customMockERC20Metadata = new MockERC20Metadata(6);
-        uint64 cosmosAmount = 0; // 1 Cosmos coin
+    // Invariant test: ERC20 -> SdkCoin with tokenDecimals > DEFAULT_COSMOS_DECIMALS
+    /**
+     * @notice Invariant test for ERC20 to Cosmos coin conversion with token decimals greater than Cosmos decimals
+     * @notice Given random inputs, this test applies invariant checking previously defined
+     * @param amount The amount of ERC20 tokens to be converted
+     * @param decimals The number of decimals of the ERC20 token
+     * Requirements:
+     * - `amount` must be greater than 0 and less than or equal to the maximum uint64 value.
+     * - `decimals` must be greater than the default Cosmos decimals (6) and less than or equal to 77.
+     */
+    function testInvariant_ERC20ToSdkCoin_BiggerDecimals(uint256 amount, uint8 decimals) public {
+        // Inputs constraints
+        if (amount == 0 || amount > ~uint64(0) || decimals <= 6 || decimals > 77) {
```

Comment:
It would be good to define these constant at the top (I mean `6` and `77`). Do this in other places they come up as well